code/FBX/FBXConverter.cpp

```diff
@@ -1566,7 +1566,7 @@ namespace Assimp {
             }
             catch (std::exception&e) {
                 std::for_each(bones.begin(), bones.end(), Util::delete_fun<aiBone>());
-                throw;
+                throw e;
```

Comment:
This is wrong. It will cause any exception which has been derived from `std::exception` (like our `DeadlyImportError` to be "sliced" back into `std::exception`, losing important information about it.

The correct way to fix this is to remove the named variable from `catch` :

```
catch (std::exception& /* e */) {
``` 

or not care about the exception type at all, like this:

```
catch (...) {
```

Reply:
One still has the exception message which is stored in `std::exception` compared to ignoring the complete exception. ¯\_(ツ)_/¯

Reply:
I though so too but it doesn't seem to work for some reason. See #2898

Reply:
Thanks, it is the first time I heard about this pattern to correctly rethrow in C++. 👍